diagnostics/stc_annular_modes/stc_annular_modes_plot.py

```diff
@@ -0,0 +1,250 @@
+        Used to determine plot characteristics such as color levels, contour
+        lines, and labels.
+
+    title: str
+        The string specifying the title to use for the plot.  Defaults to
+        an empty string.
+
+    Returns
+    -------
+    fig : `matplotlib.pyplot.figure`
+        The matplotlib figure instance for the plot.
+
+    """
+
+    def _doy_fig_params(diag):
+        r""" Nested function for setting the plot parameters.
+
+        Parameters
+        ----------
+        diag : str
+            The string specifying the diagnostic to be plotted.
+
+        Returns
+        -------
+        params : dict
+            The plot parameters for the given diagnostic.
+
+        """
+
+        if (diag == 'eftscale'):
+            clevs = [5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20,
+                     24, 28, 32, 36, 40, 48, 56, 64, 72, 80, 88]
+            clines = clevs+[96, 104, 112, 120, 130, 140, 150]
+            cbar_label = 'e-folding timescale [days]'
+            csfmt = '%d'
+        elif (diag == 'interannstdv'):
+            clevs = np.linspace(0, 2, 21)
+            clines = list(clevs)+[2.2, 2.4, 2.6, 2.8, 3.0, 3.5, 4.0]
+            cbar_label = 'Interannual Std. Deviation'
+            csfmt = '%0.1f'
+        elif (diag == 'predictability'):
+            clevs = np.linspace(0, 0.4, 21)
+            clines = list(clevs)+[0.42, 0.44, 0.46, 0.48,
+                                  0.5, 0.55, 0.6, 0.65, 0.70]
+            cbar_label = 'Predictability (Fraction of Variance explainable)'
+            csfmt = '%0.2f'
+        else:
+            msg = f'Diagnostic {diag} not supported!'
+            raise ValueError(msg)
+
+        params = {
+            'clevs': clevs,
+            'clines': clines,
+            'cblabel': cbar_label,
+            'csfmt': csfmt
+        }
+        return params
+
+    # make sure user inputs a valid diagnostic
+    diag_options = ['eftscale', 'interannstdv', 'predictability']
+    if (diag not in diag_options):
+        msg = f'diag must be one of {diag_options}'
+        raise ValueError(msg)
+
+    # get our figure params
+    fig_params = _doy_fig_params(diag)
+
+    # Figure out how we're handling the time axis. We will plot with Jan 1
+    # centered in the middle of the plot, but the position of Jan 1 and other
+    # months varies based on the underlying calendar of the data
+    max_doy = int(dat.dayofyear.max())
+    if (max_doy == 365):
+        # July 1 falls on DOY 182
+        roll_to = -181
+        xticks = np.array([1, 32, 63, 93, 124, 154, 185,
+                          213, 244, 274, 305, 335, 365])
+    elif (max_doy == 360):
+        # July 1 falls on DOY 181
+        roll_to = -180
+        xticks = np.array([1, 31, 61, 91, 121, 151, 181,
+                          211, 241, 271, 301, 331, 360])
+    else:
+        msg = "The maximum dayofyear in the input data should only be 360 or 365"
+        raise ValueError(msg)
+
```

Comment:
Will this fail if run on an experiment that has leap years?

Reply:
In my main driver script, current lines 232-236 I have

```python
# Check if the data has leapyears, and get rid of them if so
max_doy = int(zzm['time.dayofyear'].max())
if (max_doy == 366):
    print("*** Removing leapdays from data")
    zzm = zzm.convert_calendar('noleap', use_cftime=True)
```

Leapdays pose a bit of a problem for the two-step filter, since regardless of how you groupby (whether by dayofyear or by explicit "MMDD"), you may not have enough leapdays to apply the 30-year lowpass filter.